specs/networking/libp2p-standardization.md

```diff
@@ -112,21 +139,27 @@ propagation times and message duplication. Current network-related defaults are:
 updated to topic hashes in later versions - https://github.com/libp2p/rust-libp2p/issues/473*
 
 For Eth2.0 clients, topics are sent as `SHA2-256` hashes of the topic string.
+Topic strings have form: `<TopicName>/<TopicEncoding>`. The `<TopicName>`
+defines the type of data being sent on the topic and the `<TopicEncoding>`
+defines how the data field of the message is encoded. (Further details can be
+found in [Messages](#Messages)).
 
 There are two main topics used to propagate attestations and beacon blocks to
-all nodes on the network.
+all nodes on the network. Their `<TopicName>`'s are:
 
-- The `beacon_block` topic - This topic is used solely for propagating new
-	beacon blocks to all nodes on the networks.
-- The `beacon_attestation` topic - This topic is used to propagate
-	aggregated attestations to subscribing nodes (typically block proposers) to
-	be included into future blocks. Attestations are aggregated in their
-	respective subnets before publishing on this topic.
+- `beacon_block` - This topic is used solely for propagating new
+	beacon blocks to all nodes on the networks. Blocks are sent in their
+	entirety. Clients who receive blocks on a topic SHOULD validate the block
+	proposer signature before forwarding it across the network.
+- `beacon_attestation`  - This topic is used to propagate
+	aggregated attestations (in their entirety) to subscribing nodes (typically
+	block proposers) to be included into future blocks. Attestations are
+	aggregated in their respective subnets before publishing on this topic.
 
 Shards are grouped into their own subnets (defined by a shard topic). The
 number of shard subnets is defined via `SHARD_SUBNET_COUNT` and the shard
 `shard_number % SHARD_SUBNET_COUNT` is assigned to the topic:
-`shard{shard_number % SHARD_SUBNET_COUNT}_attestation`.
+`shard{shard_number % SHARD_SUBNET_COUNT}_beacon_attestation`.
```

Comment:
How will node advertise a shard topic that it is assigned to? A straightforward solution would be to use ENR records for this purpose. If this is what we want then maybe it worth listing under ENR records of Discovery section?

Reply:
I may be missing something. If a validator is assigned to attest to a shard, it should subscribe and publish on that shard topic. 

If in phase 1 a node wants to communicate blocks and messages dedicated to that shard, I imagine we will have a specific shard topic for that. This scenario is not included in this phase 0 spec however.

Reply:
From [gossipsub](https://github.com/libp2p/specs/tree/master/pubsub/gossipsub#topic-membership) spec:
> Also note that as part of the pubsub api, the peer emits `SUBSCRIBE` and `UNSUBSCRIBE` control messages to all its peers whenever it joins or leaves a topic. This is provided by the the ambient peer discovery mechanism and nominally not part of the router. A standalone implementation would have to implement those control messages.

We need a mechanism that spreads node's subscription updates all over the network. From my understanding Discovery v5 was designed bearing capability updates in mind.

> This scenario is not included in this phase 0 spec however.

Depending on aggregation strategy and number of peers in the network subnets can become a unit of aggregation process in phase 0.